core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -213,6 +213,11 @@ private[spark] class MemoryStore(blockManager: BlockManager, memoryManager: Memo
   }
 
   override def remove(blockId: BlockId): Boolean = memoryManager.synchronized {
+    val referenceCount = blockManager.getReferenceCount(blockId)
```

Comment:
What are the semantics here? It seems reasonable for another thread to get this block. Who calls remove?